### ISSUE TITLE
Sync partners cannot open synced files (corrupted in sync process), Sabre\DAV\Exception\Forbidden, <s:message>Unsupported client version.</s:message

### ISSUE THREAD

**Issue by tosod** [owncloud/client#3925]
Steps to reproduce
------------
* Folder shared with several clients 
* Most clients running version 2.0.1.5446, a problematic client running 1.7.. (most likely 1.7.1 (build 4382)) (+ potentially one or two running version 1.8.0.4893 also)
* Server "version":"8.1.1.3","versionstring":"8.1.1"

Expected behaviour
------------
Files sync successfully from the 1.7.1 (build 4382) client without being corrupted in the process

Actual behaviour
------------
Files created at the 1.7.1 (build 4382) client become corrupted in the syncing process and cannot be opened at sync partners computers. Happens for several different filetypes This happens to .doc, .pdf, .ppt or .pptx, at the very least.

Opening a synced word document from a synked computer simply returns this:
```
The document cannot be opened because there are problems with the content. 
Details: An invalid character was found in text content.
```

If I open it with the option to recover text from any file I find this:
```
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\Forbidden</s:exception>
  <s:message>Unsupported client version.</s:message>
</d:error>
```
Updating the 1.7.1 (build 4382) client to version 2.0.1.5446 at the problematic computer does not solve the issue with the files at other computers which already synced them.

The files are still intact at the problematic client they originated from and open without problem at that computer.

**Notes**
Backwards compatibility should perhaps be checked and possibly automatic detection of the issue somehow or an option to manually force a overwrite full resync with the problematic client as athorative sync member would be helpful is my suggestion.

Do you want any logs to look into the issue? I possibly might still aqquire them.

**Update 5.11** 
Problem happened on at least 2 machines.
**wrong info:** *Files seem to be ok at the server, seems to be corrupted when syncing to the  client version 2.0.1.5446 from the server.*

**Comment by guruz** [comment]
@tosod Not sure I understand completely.
Are you sure the server has the correct file?
Does a 2.x client download a file that then has those XML contents? 

FYI @LukasReschke

**Comment by ogoffart** [comment]
Normaly we should not save the file if the server returns an error.
It only keeps the files if the server returns 200 OK, and a ETag header.

So does that mean the server had been sending these exception on a 200 OK?

**Comment by DeepDiver1975** [comment]
this could be related to the order of plugins to be executed in the server

@PVince81 @MorrisJobke @rullzer @nickvergessen we did some changes recently - related?

**Comment by PVince81** [comment]
I checked `% git diff stable7..v8.1.3 remote.php`

It seems the order of plugins did not change.

**Comment by DeepDiver1975** [comment]
if I recall properly we changed that in 8.2 or master to fix some similar issue in this area - if I recall correctly

**Comment by nickvergessen** [comment]
Well not sure, but for the maintenance plugin I had to make two patches, because [changing the order in the file helped in 8.1](https://github.com/owncloud/core/pull/19075/files), while for master I had to [change the priority of the plugin](https://github.com/owncloud/core/pull/19095/files). So maybe something related to the randomness of sabredav.

**Comment by guruz** [comment]
@tosod Any new info from your side?

**Comment by ogoffart** [comment]
But what is throwing `Unsupported client version.`  error?  Can it be thrown after the 200 OK and E-Tag was sent?

**Comment by nickvergessen** [comment]
The `Unsupported client version.` error means that the client is not accepted by the server because it is too old (< v1.7.0). So no, that should not happen after an E-Tag or 200 or what ever?

**Comment by tosod** [comment]
@guruz Well here's a small update for now:

I wrote some info that was wrong in the original post it seems, sorry about that.
1) The files synced to the server from the old client computer are corrupted server-side too and do not open even in the website. I got wrong info from an user earlier.
2) Files synced from newer clients do not open from the old client computer.

What is interesting is that the old client is allowed to sync at all, it should be forbidden from syncing with the server, right?

I tried checking the logs of the server but regretfully the log file had gotten full and didn't log during that time period, I will try and check a newer synced file that is corrupted tomorrow if I get the opportunity.

We have one problematic client left to my knowledge, I'm hoping to be able to access the client tomorrow and fetch logs from it and get the exact owncloud version number.

If it is inything in particular you think I should check for please do let me know.

**Comment by ogoffart** [comment]
I changed the user agent of my client to something old.  And it won't let me login (saying the client is unsuported).  So there is no way it can download or upload anything.

I have no idea how the files could have been corrupted.

**Comment by ogoffart** [comment]
Is there any way this can be reproduced?

**Comment by tosod** [comment]
@ogoffart 

Every other client I've seen reports same as your, some message something about unsupported client and is blocked from syncing. However that particular one did not apparently. I think I've even seen the same version clients which were blocked (since we deployed only a few versions of the clients in batches some were bound to have been the same) so I don't know what to think about it.

I tried checking the server log but regretfully it seems it was full and was unable to write anything at the time during vacation time when this issue occured, too bad :(. It did not contain any useful information.

After updating the client and server the issue has not reoccured.. Unless anybody else is able to reproduce the issue I will consider the issue as resolved after updating on my part. Hopefully this will never occur for anybody again. I'll let you guys decide whether to close the case or not.
